landlab/components/bedrock_landslider/bedrock_landslider.py

```diff
@@ -0,0 +1,692 @@
+
+        if fraction_fines_LS > 1.0 or fraction_fines_LS < 0.0:
+            raise ValueError(
+                f"Fraction of fines must be between 0 and 1 ({fraction_fines_LS})"
+            )
+
+        # Set seed
+        if seed is not None:
+            np.random.seed(seed)
+
+    def _landslide_erosion(self, dt):
+        """
+        Calculate bedrock landsliding for a time period 'dt'.
+
+        Parameters
+        ----------
+        dt: float (time)
+            The imposed timestep.
+
+        Returns
+        -------
+        suspended_Sed : float
+            Volume of suspended sediment.
+
+        """
+        # Pointers
+        topo = self.grid.at_node["topographic__elevation"]
+        bed = self.grid.at_node["bedrock__elevation"]
+        ss = self.grid.at_node["topographic__steepest_slope"]
+        soil_d = self.grid.at_node["soil__depth"]
+        ls_sed_in = self.grid.at_node["landslide_sediment_point_source"]
+        lb_bed_ero = self.grid.at_node["landslide__bed_erosion"]
+
+        # Reset LS Plains
+        lb_bed_ero[:] = np.zeros(self.grid.number_of_nodes)
+        # Reset landslide sediment point source field
+        ls_sed_in[:] = np.zeros(self.grid.number_of_nodes)
+
+        # Identify flooded nodes
+        flood_status = self.grid.at_node["flood_status_code"]
+        flooded_nodes = np.nonzero(flood_status == _FLOODED)[0]
+
+        """In the following section the location of critical nodes where
+        landsldies are initatated is calcualted, unless these critical nodes
+        are provided as critical_sliding_nodes"""
+        if self.critical_sliding_nodes is None:
+            # Calculate gradients
+            H_el = topo - topo[self.grid.at_node["flow__receiver_node"]]
+
+            H_el[flooded_nodes] = 0
+            H_el[H_el > MAX_HEIGHT_SLOPE] = MAX_HEIGHT_SLOPE
+
+            sc_rad = np.arctan(self._angle_int_frict)
+            Hc = (4 * self._cohesion_eff / (self._grav * self._rho_r)) * (
+                (np.sin(np.arctan(ss)) * np.cos(sc_rad))
+                / (1 - np.cos(np.arctan(ss) - sc_rad))
+            )
+            Hc[Hc == 0] = np.nan
+            # Spatial probability
+            p_S = H_el / Hc
+            p_S[np.arctan(ss) <= sc_rad] = 0
+            p_S[p_S > 1] = 1
+
+            # Temporal probability
+            p_t = 1 - np.exp(-dt / self._landslides_return_time)
+
+            # Combined probability
+            p = p_t * p_S
+            slides = np.random.rand(p.size) < p
+
+            # Now, find the critical node, which is the receiver of i_slide
+            # Critical nodes must be unique (a given node can have more receivers...)
+            i_slide = np.unique(
+                self.grid.at_node["flow__receiver_node"][np.where(slides)]
+            )
+            # Remove boundary nodes
+            if not self.landslides_on_boundary_nodes:
+                i_slide = i_slide[(self.grid.node_is_boundary(i_slide) is False)]
```

Comment:
I'm not sure this line does what you want it to do (it doesn't look like it's tested). I think what you want is the following?
```python
i_slide = i_slide[~self.grid.node_is_boundary(i_slide)]
```

Reply:
Nicely spotted!